krkn/scenario_plugins/node_actions/gcp_node_scenarios.py

```diff
@@ -219,14 +211,14 @@ def node_stop_scenario(self, instance_kill_count, node, timeout):
         for _ in range(instance_kill_count):
             try:
                 logging.info("Starting node_stop_scenario injection")
-                instance_id, zone = self.gcp.get_instance_id(node)
+                instance_name, zone = self.gcp.get_instance_name(node)
                 logging.info(
-                    "Stopping the node %s with instance ID: %s " % (node, instance_id)
+                    "Stopping the node %s with instance ID: %s " % (node, instance_name)
                 )
-                self.gcp.stop_instances(zone, instance_id)
-                self.gcp.wait_until_stopped(zone, instance_id, timeout)
+                self.gcp.stop_instances(zone, instance_name)
+                self.gcp.wait_until_stopped(zone, instance_name, timeout)
                 logging.info(
-                    "Node with instance ID: %s is in stopped state" % instance_id
+                    "Node with instance ID: %s is in stopped state" % instance_name
                 )
                 nodeaction.wait_for_unknown_status(node, timeout, self.kubecli)
```

Comment:
Not sure how, but think the first parameter here needs to be _instance_name_, the script gets stuck for a while but progresses, no clear error message though and node is not ready pretty quickly so not sure why its waiting so long


```
2024-10-23 22:05:28,533 [INFO] Node with instance ID: prubenda-4185-c8qj7-worker-b-mqtcq is in stopped state


2024-10-23 22:11:28,653 [INFO] Waiting for 20 seconds before starting the node
2024-10-23 22:11:48,658 [INFO] Starting node_start_scenario injection
2024-10-23 22:11:54,235 [INFO] Starting the node prubenda-4185-c8qj7-worker-b-mqtcq with instance ID: prubenda-4185-c8qj7-worker-b-mqtcq 

```

Reply:
Can you check with the newly pushed code? I haven't found this issue in my testing.

Reply:
Think this might be an issue with the wait function. We can tackle if it continues to be an issue

Reply:
Created this PR that I think will fix the issue i was seeing: https://github.com/krkn-chaos/krkn-lib/pull/134